### ISSUE TITLE
CodeBuild doesn't wait long enough for stack updates to complete

### ISSUE THREAD

**Issue by jcaxmacher** [awslabs/aws-deployment-framework#240]
In places where the ADF deploys CloudFormation templates from CodeBuild (in the master account for bootstrap templates and in the deployment account for creating pipelines), it sometimes fails because the waiter is not waiting long enough.  [This code](https://github.com/awslabs/aws-deployment-framework/blob/master/src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/cloudformation.py#L141) waits a random amount of time which could be as little as 8.25 minutes (45 periods * 11 seconds).  I'd like to see this be configurable or have the default minimum increased.

Here's the error from CodeBuild in our master account:

```
2020-02-05 14:14:50,539 | INFO | cloudformation | 320398254626 - Executing Cloudformation Change Set with name: adf-global-base-master | (cloudformation.py:252)
2020-02-05 14:14:50,722 | INFO | cloudformation | 320398254626 - Waiting for CloudFormation stack: adf-global-base-master in us-east-1 to reach stack_update_complete | (cloudformation.py:137)
Traceback (most recent call last):
  File "adf-build/main.py", line 348, in <module>
    main()
  File "adf-build/main.py", line 326, in main
    thread.join()
  File "/codebuild/output/src548028093/src/adf-build/shared/python/thread.py", line 30, in join
    raise self.exc
  File "/codebuild/output/src548028093/src/adf-build/shared/python/thread.py", line 22, in run
    **self._kwargs
  File "adf-build/main.py", line 218, in worker_thread
    cloudformation.create_stack()
  File "/codebuild/output/src548028093/src/adf-build/shared/python/cloudformation.py", line 279, in create_stack
    self._execute_change_set(waiter)
  File "/codebuild/output/src548028093/src/adf-build/shared/python/cloudformation.py", line 259, in _execute_change_set
    self._wait_stack(waiter, self.stack_name)
  File "/codebuild/output/src548028093/src/adf-build/shared/python/cloudformation.py", line 143, in _wait_stack
    'MaxAttempts': 45
  File "/usr/local/lib/python3.7/site-packages/botocore/waiter.py", line 53, in wait
    Waiter.wait(self, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/botocore/waiter.py", line 323, in wait
    last_response=response,
botocore.exceptions.WaiterError: Waiter StackUpdateComplete failed: Waiter encountered a terminal failure state

[Container] 2020/02/05 14:15:20 Command did not exit successfully python adf-build/main.py exit status 1
```

**Comment by bundyfx** [comment]
Thanks @jcaxmacher 

Indeed we should bump the default time up a bit, this will go into 3.0.5 as a minor change.

**Comment by sbkok** [comment]
I'm closing this issue as it has been inactive for a long time.  This probably means that it is not reproducible or it has been fixed in the meanwhile with version [v3.0.6](https://github.com/awslabs/aws-deployment-framework/releases/tag/v3.0.6).

Please reopen if you still encounter this issue with the [latest stable version](https://github.com/awslabs/aws-deployment-framework/releases).

Thank you!
